File: fc-chat/demo/part3/ch2/05-session-using-db-redis/src/main/java/net/prostars/messagesystem/config/RedisSessionConfig.java
package net.prostars.messagesystem.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.redis.serializer.GenericJackson2JsonRedisSerializer;
import org.springframework.data.redis.serializer.RedisSerializer;
import org.springframework.security.jackson2.SecurityJackson2Modules;
import org.springframework.session.data.redis.config.annotation.web.http.EnableRedisHttpSession;

@Configuration
@EnableRedisHttpSession(redisNamespace = "message:user_session", maxInactiveIntervalInSeconds = 300)
@SuppressWarnings("unused")
public class RedisSessionConfig {

  @Bean
  public RedisSerializer<Object> springSessionDefaultRedisSerializer() {
    ObjectMapper mapper = new ObjectMapper();
    mapper.registerModules(SecurityJackson2Modules.getModules(getClass().getClassLoader()));
    return new GenericJackson2JsonRedisSerializer(mapper);
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/config/RedisSessionConfig.kt
package com.yscorp.withpush.messagesystem.config

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.redis.serializer.GenericJackson2JsonRedisSerializer
import org.springframework.data.redis.serializer.RedisSerializer
import org.springframework.security.jackson2.SecurityJackson2Modules
import org.springframework.session.FlushMode
import org.springframework.session.data.redis.config.annotation.web.http.EnableRedisHttpSession

@Configuration
@EnableRedisHttpSession(
    redisNamespace = "message:user_session",
    maxInactiveIntervalInSeconds = 300,
    flushMode = FlushMode.IMMEDIATE
)
@Suppress("unused")
class RedisSessionConfig {
    @Bean
    fun springSessionDefaultRedisSerializer(): RedisSerializer<Any> {
        val mapper = ObjectMapper()
        mapper.registerModules(SecurityJackson2Modules.getModules(javaClass.classLoader))
        return GenericJackson2JsonRedisSerializer(mapper)
    }
}
